src/challenge/Challenge.tsx

```diff
@@ -1,523 +1,637 @@
-import React, { MutableRefObject } from "react";
-import { throttle } from "lodash";
-import { Box, Card, CardContent, Grid, IconButton, Paper } from "@mui/material";
-import CachedIcon from "@mui/icons-material/Cached";
+import React, {
```

Comment:
Complete rewrite using functional React component, merging editor and normal challenge, delegating much of the challenge class's responsibility